From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MichaIng <micha@dietpi.com>
Date: Sun, 2 Aug 2026 18:58:58 +0200
Subject: [ARCHEOLOGY] orangepizero2w/zero3: add dvfs patch more cleanly

> X-Git-Archeology: > recovered message: > Skip all the redundant stuff from the old patch. Just add the delays to the existing regulator nodes.
> X-Git-Archeology: > recovered message: > Apply patch to Orange Pi Zero 2W as well, which suffers from the same stability issues when the uwe5622 driver is loaded.
> X-Git-Archeology: > recovered message: > Signed-off-by: MichaIng <micha@dietpi.com>
> X-Git-Archeology: - Revision de29319443c007fb968eb4aec96faaf565534d37: https://github.com/armbian/build/commit/de29319443c007fb968eb4aec96faaf565534d37
> X-Git-Archeology:   Date: Sun, 02 Aug 2026 18:58:58 +0200
> X-Git-Archeology:   From: MichaIng <micha@dietpi.com>
> X-Git-Archeology:   Subject: orangepizero2w/zero3: add dvfs patch more cleanly
> X-Git-Archeology:
---
 arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts | 7 +++++++
 arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts  | 7 +++++++
 2 files changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts b/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero2w.dts
@@ -292,6 +292,8 @@ reg_dcdc1: dcdc1 {
 				regulator-min-microvolt = <810000>;
 				regulator-max-microvolt = <990000>;
 				regulator-name = "vdd-gpu-sys";
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
 			};
 
 			reg_dcdc2: dcdc2 {
@@ -299,6 +301,9 @@ reg_dcdc2: dcdc2 {
 				regulator-min-microvolt = <810000>;
 				regulator-max-microvolt = <1100000>;
 				regulator-name = "vdd-cpu";
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
+				regulator-ramp-delay = <200>;
 			};
 
 			reg_dcdc3: dcdc3 {
@@ -306,6 +311,8 @@ reg_dcdc3: dcdc3 {
 				regulator-min-microvolt = <1100000>;
 				regulator-max-microvolt = <1100000>;
 				regulator-name = "vdd-dram";
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
 			};
 		};
 	};
diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts b/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts
@@ -82,6 +82,8 @@ reg_dcdc1: dcdc1 {
 				regulator-min-microvolt = <810000>;
 				regulator-max-microvolt = <990000>;
 				regulator-name = "vdd-gpu-sys";
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
 			};
 
 			reg_dcdc2: dcdc2 {
@@ -89,6 +91,9 @@ reg_dcdc2: dcdc2 {
 				regulator-min-microvolt = <810000>;
 				regulator-max-microvolt = <1100000>;
 				regulator-name = "vdd-cpu";
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
+				regulator-ramp-delay = <200>;
 			};
 
 			reg_dcdc3: dcdc3 {
@@ -96,6 +101,8 @@ reg_dcdc3: dcdc3 {
 				regulator-min-microvolt = <1100000>;
 				regulator-max-microvolt = <1100000>;
 				regulator-name = "vdd-dram";
+				regulator-step-delay-us = <25>;
+				regulator-final-delay-us = <50>;
 			};
 		};
 	};
-- 
Armbian

